http-server/src/tests.rs

```diff
@@ -380,6 +381,27 @@ async fn can_register_modules() {
 #[tokio::test]
 async fn stop_works() {
 	let _ = env_logger::try_init();
-	let (_addr, stop_handle) = server().with_default_timeout().await.unwrap();
-	assert!(matches!(stop_handle.stop().unwrap().await, Ok(_)));
+	let (_addr, server_handle) = server().with_default_timeout().await.unwrap();
+	assert!(matches!(server_handle.stop().unwrap().await, Ok(_)));
+}
+
+#[tokio::test]
+async fn run_forever() {
+	const TIMEOUT: Duration = Duration::from_millis(200);
+
+	let _ = env_logger::try_init();
+	let (_addr, server_handle) = server().with_default_timeout().await.unwrap();
+	// Timed out.
+	server_handle.with_timeout(TIMEOUT).await.unwrap_err();
+
+	let (_addr, server_handle) = server().await;
+	server_handle.handle.as_ref().unwrap().abort();
```

Comment:
ok, you did this instead of `stop` because it moves the `JoinHandle`?

Reply:
Stop is a little bit different, here we want to emulate the actual server behavior -- if a task is cancelled or panics, the future handle resolves too.